.evergreen/config/build/build-libmongoc-8.2.yml

```diff
@@ -0,0 +1,27 @@
+# This file is generated automatically - please edit the corresponding template file!
```

Comment:
The comment is sensible, but what do you think about also nesting these generated files under a sub-directory? That's done in libmongoc (see: [.evergreen/generated_configs](https://github.com/mongodb/mongo-c-driver/tree/master/.evergreen/generated_configs)).

I think this makes sense if we never expect to work with the generated files directly, in which case we can just stuff them away. That's in contrast to the current approach where you're primarily grouping by "build" and "test" types.

Consider the following directory structure:

```
.evergreen/config/
├─ build-task-groups.yml
├─ build-variants.yml
├─ functions.yml
├─ generate-config.php
├─ test-tasks.yml
├─ test-variants.yml
├─ generated/
│  ├─ build-libmongoc-8.2.yml
│  ├─ build-php-7.4.yml
│  ├─ test-load-balanced-5.0.yml
│  ├─ ...
├─ templates/
│  ├─ build-libmongoc-template.yml
│  ├─ build-php-template.yml
│  ├─ test-load-balanced-template.yml
│  ├─ ...
```

Alternatively, we could avoid a `templates/` directory and just keep the `_template-foo.yml` files directly within `.evergreen/config/`, which would then be a central directory of files we might need to modify over time.

I also added a "test-" prefix on generated test configs above to segregate them from build files, but I suppose that's not really important if we don't expect naming conflicts.

You could also replace "build-" and "test-" prefixes with `build/` and `test/` subdirectories within `generated/`; however, I don't think there's strong justification for that if we'd generally have no reason to venture into `generated/` anyway.

I don't mean to bikeshed here, but more thinking about how we can avoid having to look at files we'd have no business touching.

Reply:
Split up into `generated` and `templates` as suggested, but kept the `build` and `test` folders for better distinction.